sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1901,6 +1909,7 @@ case class NestedStruct(f: ClassData)
 case class DeepNestedStruct(f: NestedStruct)
 
 case class InvalidInJava(`abstract`: Int)
+case class InvalidInJavaForNumber(`1something`: String)
```

Comment:
Looks like it makes doc generation fail, though honestly I'm not aware of how to deal with  this.

Reply:
If the doc generation is in java it has the same problem, because it starts with a number, and it is not allowed. I don't know if I can help with it.

Reply:
Could you add `private[sql]` and verify it doesn't make the test in local, and push to trigger new build? Non-public classes will be excluded if I understand correctly.